src/core/include/rs/feature_structure_proxy.h

```diff
@@ -662,6 +697,42 @@ class ListFeatureStructureEntry : public FeatureStructureEntry<std::vector<T, TA
     return success;
   }
 
+  template<typename TargetT, typename AnnotT>
+  bool filter(std::vector<TargetT> &result, std::vector<std::vector<AnnotT>> &annots)
```

Comment:
as the two functions have the same functionality, only one time for a list and one time for an array, here is one example for how to use the function.

```
std::vector<rs::Cluster> toolC;
std::vector<std::vector<percepteros::ToolObject>> toolA;

scene.identifiables.filter(rackC, rackA);
```

Explanation of code lines:

1. std::vector<rs::Cluster> toolC;
This is the vector containing all the clusters that are found in the containing list that have the specified kind of annotation. Generally the type used for the result vector can be of any kind, but in order to use the functionality of the function it has to be a type that can contain annotations - otherwise nothing will be returned.

2. std::vector<std::vector<percepteros::ToolObject>> toolA;
This is the vector of vectors for all the annotations of the clusters. It has to be a vector of vectors since every cluster can have multiple annotations of the same type. The vector of annotations at index 0 corresponds to the cluster at index 0 and so on. As with the clusters above, the type of the vectors can be anything, but in order to get a result they have to be some type of annotations.

3. scene.identifiables.filter(rackC, rackA);
This is the call, filtering out all of the clusters containing annotations of type ToolAnnotations and the annotations they have from the list scene.identifiables